### ISSUE TITLE
no-explicit-any ignores generic in function defintion

### ISSUE THREAD

**Issue by MayhemYDG** [bradzacher/eslint-plugin-typescript#279]
**Repro**

```TS
function test<T extends Partial<any>>() {}

const test = <T extends Partial<any>>() => {};
```

**Expected Result**
`any` should not be allowed there with the rule `no-explicit-any` enabled.
**Actual Result**
`any` is not detected.

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `0.14.0` |
| `typescript-eslint-parser` | `21.0.2` |
| `typescript`               | `3.2.2` |

**Comment by armano2** [comment]
Issue is related to parser, and it got fixed in one of prs
